platforms/common/src/CommonKernels.cpp

```diff
@@ -8079,18 +8079,63 @@ void CommonCalcATMForceKernel::initialize(const System& system, const ATMForce&
     numParticles = force.getNumParticles();
     if (numParticles == 0)
         return;
-    vector<mm_float4> displVector1(cc.getPaddedNumAtoms(), mm_float4(0, 0, 0, 0));
-    vector<mm_float4> displVector0(cc.getPaddedNumAtoms(), mm_float4(0, 0, 0, 0));
-    for (int i = 0; i < numParticles; i++) {
-        Vec3 displacement1, displacement0;
-        force.getParticleParameters(i, displacement1, displacement0);
-        displVector1[i] = mm_float4(displacement1[0], displacement1[1], displacement1[2], 0);
-        displVector0[i] = mm_float4(displacement0[0], displacement0[1], displacement0[2], 0);
-    }
-    displ1.initialize<mm_float4>(cc, cc.getPaddedNumAtoms(), "displ1");
-    displ1.upload(displVector1);
-    displ0.initialize<mm_float4>(cc, cc.getPaddedNumAtoms(), "displ0");
-    displ0.upload(displVector0);
+    vector<int>  pj1Vector(cc.getPaddedNumAtoms());
+    vector<int>  pi1Vector(cc.getPaddedNumAtoms());
+    vector<int>  pj0Vector(cc.getPaddedNumAtoms());
+    vector<int>  pi0Vector(cc.getPaddedNumAtoms());
```

Comment:
Using four separate arrays like this will lead to inefficient memory access in the kernels.  If you pack them into a single int4 array, it will be faster.

Reply:
Good point, I will do that.